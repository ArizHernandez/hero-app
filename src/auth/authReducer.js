import { types } from "../types/types";

/* State 
{
  name: 'Marco',
  isLoged: true/false
}
*/ 

export const authReducer = ( state = {}, action ) => {

  switch ( action.type ) {
    case types.login:
        return { 
          ...action.payload, 
          isLoged: true 
        }
    case types.logout:
      return {
        isLoged: false
      }
  
    default:
      return state;
  }
}