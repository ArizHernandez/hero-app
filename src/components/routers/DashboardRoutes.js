import React from 'react'
import { Redirect, Route, Switch } from 'react-router'

import { Navbar } from '../ui/Navbar'
import { DcScreen } from '../dc/DcScreen'
import { HeroScreen } from '../heroes/HeroScreen'
import { MarvelScreen } from '../Marvel/MarvelScreen'
import { SearchScreen } from '../search/SearchScreen'

export const DashboardRoutes = () => {
  return (
    <>
      <Navbar/>

      <div>
        <Switch>
          <Route exact path="/marvel" component={ MarvelScreen }/>
          <Route exact path="/dc" component={ DcScreen }/>
          <Route exact path="/hero/:heroId" component={ HeroScreen }/>
          <Route exact path="/search" component={ SearchScreen }/>
          <Redirect to="/marvel"/>
        </Switch>
      </div>
    </>
  )
}
