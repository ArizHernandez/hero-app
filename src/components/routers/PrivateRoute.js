import React from 'react'
import PropTypes from 'prop-types';
import { Redirect, Route } from 'react-router'

export const PrivateRoute = ({ 
  component: Component,
  isLoged,
  ...rest
}) => {

  localStorage.setItem( 'lastPath', rest.location.pathname + rest.location.search );

  return (
    <Route { ...rest }
      component={ ( props ) => (
        isLoged 
        ? ( <Component { ...props }/> )
        : ( <Redirect to="/login"/> )
      )}
    />
  )
}

PrivateRoute.propTypes = {
  isLoged: PropTypes.bool.isRequired,
  component: PropTypes.func.isRequired
}