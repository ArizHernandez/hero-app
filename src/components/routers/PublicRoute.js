import React from 'react'
import { Route, Redirect } from 'react-router'

export const PublicRoute = ({
  component: Component,
  isLoged,
  ...rest
}) => {

  return (
    <Route
      { ...rest }
      component={ ( props ) => (
        !isLoged 
        ? ( <Component { ...props }/> )
        : ( <Redirect to="/"/>)
      )}
    />
  )
}
