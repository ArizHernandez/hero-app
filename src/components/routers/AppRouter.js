import React, { useContext } from 'react';
import {
  BrowserRouter as Router,
  Switch,
} from "react-router-dom";
import { AuthContext } from '../../auth/AuthContext';
import { LoginScreen } from '../login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {

  const { user: { isLoged } } = useContext(AuthContext);

  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute 
            exact 
            isLoged={ isLoged }
            path="/login" 
            component={ LoginScreen }
          />
          <PrivateRoute 
            path="/" 
            isLoged={ isLoged }
            component={ DashboardRoutes }
          />
        </Switch>
      </div>
    </Router>
  )
}
