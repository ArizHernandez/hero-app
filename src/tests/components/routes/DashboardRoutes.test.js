import { mount } from "enzyme"
import { MemoryRouter } from "react-router"
import { AuthContext } from "../../../auth/AuthContext"
import { DashboardRoutes } from "../../../components/routers/DashboardRoutes"

describe('Pruebas al <DasboardRoutes />', () => {
  
  const context = {
    user:{
      name: 'Ariz',
      isLoged: true
    },
    dispatch: jest.fn()
  }

  test('Debe de cargar todos los componentes correctamente', () => {
    const wrapper = mount(
      <AuthContext.Provider value={context}>
        <MemoryRouter>
          <DashboardRoutes />
        </MemoryRouter>
      </AuthContext.Provider>
    )
      
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.text-info').text().trim()).toBe(context.user.name);
  })
  
})
