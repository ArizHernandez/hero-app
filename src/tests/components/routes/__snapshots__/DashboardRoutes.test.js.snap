// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al <DasboardRoutes /> Debe de cargar todos los componentes correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Asociaciones
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav me-auto mb-2 mb-lg-0 w-100"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/marvel"
              onClick={[Function]}
              style={Object {}}
            >
              Marvel
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/dc"
              onClick={[Function]}
            >
              DC
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/search"
              onClick={[Function]}
            >
              Search
            </a>
          </li>
          <li
            className="nav-item ms-md-auto d-flex"
          >
            <span
              className="nav-link text-info"
            >
              Ariz
            </span>
            <button
              className="nav-link btn"
              onClick={[Function]}
            >
              Logout
            </button>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div>
    <div
      className="container mt-3"
    >
      <h1
        className="text-center"
      >
        Marve Screen
      </h1>
      <hr
        className="mt-0 mb-4"
      />
      <div
        className="card-group"
      >
        <div
          className="row row-cols-1 row-cols-md-3 row-cols-lg-5 g-4"
        >
          <div
            className=" animate__animated animate__fadeIn"
            key="marvel-spider"
          >
            <div
              className="card text-center"
            >
              <img
                alt="..."
                className="card-img-top"
                src=""
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Spider Man
                </h5>
                <p
                  className="card-text"
                >
                  Peter Parker
                </p>
                <p
                  className="card-text text-end mb-2"
                >
                  <small
                    className="text-muted"
                  >
                    Amazing Fantasy #15
                  </small>
                </p>
                <a
                  className="btn btn-dark btn-block"
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
          <div
            className=" animate__animated animate__fadeIn"
            key="marvel-captain"
          >
            <div
              className="card text-center"
            >
              <img
                alt="..."
                className="card-img-top"
                src=""
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Captain America
                </h5>
                <p
                  className="card-text"
                >
                  Steve Rogers
                </p>
                <p
                  className="card-text text-end mb-2"
                >
                  <small
                    className="text-muted"
                  >
                    Captain America Comics #1
                  </small>
                </p>
                <a
                  className="btn btn-dark btn-block"
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
          <div
            className=" animate__animated animate__fadeIn"
            key="marvel-iron"
          >
            <div
              className="card text-center"
            >
              <img
                alt="..."
                className="card-img-top"
                src=""
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Iron Man
                </h5>
                <p
                  className="card-text"
                >
                  Tony Stark
                </p>
                <p
                  className="card-text text-end mb-2"
                >
                  <small
                    className="text-muted"
                  >
                    Tales of Suspense #39
                  </small>
                </p>
                <a
                  className="btn btn-dark btn-block"
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
          <div
            className=" animate__animated animate__fadeIn"
            key="marvel-thor"
          >
            <div
              className="card text-center"
            >
              <img
                alt="..."
                className="card-img-top"
                src=""
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Thor
                </h5>
                <p
                  className="card-text"
                >
                  Thor Odinson
                </p>
                <p
                  className="card-text text-end mb-2"
                >
                  <small
                    className="text-muted"
                  >
                    Journey into Myster #83
                  </small>
                </p>
                <a
                  className="btn btn-dark btn-block"
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
          <div
            className=" animate__animated animate__fadeIn"
            key="marvel-hulk"
          >
            <div
              className="card text-center"
            >
              <img
                alt="..."
                className="card-img-top"
                src=""
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Hulk
                </h5>
                <p
                  className="card-text"
                >
                  Bruce Banner
                </p>
                <p
                  className="card-text text-end mb-2"
                >
                  <small
                    className="text-muted"
                  >
                    The Incredible Hulk #1
                  </small>
                </p>
                <a
                  className="btn btn-dark btn-block"
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
          <div
            className=" animate__animated animate__fadeIn"
            key="marvel-wolverine"
          >
            <div
              className="card text-center"
            >
              <img
                alt="..."
                className="card-img-top"
                src=""
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Wolverine
                </h5>
                <p
                  className="card-text"
                >
                  James Howlett
                </p>
                <p
                  className="card-text text-end mb-2"
                >
                  <small
                    className="text-muted"
                  >
                    The Incredible Hulk #180
                  </small>
                </p>
                <a
                  className="btn btn-dark btn-block"
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
          <div
            className=" animate__animated animate__fadeIn"
            key="marvel-daredevil"
          >
            <div
              className="card text-center"
            >
              <img
                alt="..."
                className="card-img-top"
                src=""
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Daredevil
                </h5>
                <p
                  className="card-text"
                >
                  Matthew Michael Murdock
                </p>
                <p
                  className="card-text text-end mb-2"
                >
                  <small
                    className="text-muted"
                  >
                    Daredevil #1
                  </small>
                </p>
                <a
                  className="btn btn-dark btn-block"
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
          <div
            className=" animate__animated animate__fadeIn"
            key="marvel-hawkeye"
          >
            <div
              className="card text-center"
            >
              <img
                alt="..."
                className="card-img-top"
                src=""
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Hawkeye
                </h5>
                <p
                  className="card-text"
                >
                  Clinton Francis Barton
                </p>
                <p
                  className="card-text text-end mb-2"
                >
                  <small
                    className="text-muted"
                  >
                    Tales of Suspense #57
                  </small>
                </p>
                <a
                  className="btn btn-dark btn-block"
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
          <div
            className=" animate__animated animate__fadeIn"
            key="marvel-cyclops"
          >
            <div
              className="card text-center"
            >
              <img
                alt="..."
                className="card-img-top"
                src=""
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Cyclops
                </h5>
                <p
                  className="card-text"
                >
                  Scott Summers
                </p>
                <p
                  className="card-text text-end mb-2"
                >
                  <small
                    className="text-muted"
                  >
                    X-Men #1
                  </small>
                </p>
                <a
                  className="btn btn-dark btn-block"
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
          <div
            className=" animate__animated animate__fadeIn"
            key="marvel-silver"
          >
            <div
              className="card text-center"
            >
              <img
                alt="..."
                className="card-img-top"
                src=""
              />
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Silver Surfer
                </h5>
                <p
                  className="card-text"
                >
                  Norrin Radd
                </p>
                <p
                  className="card-text text-end mb-2"
                >
                  <small
                    className="text-muted"
                  >
                    The Fantastic Four #48
                  </small>
                </p>
                <a
                  className="btn btn-dark btn-block"
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  Mas...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
